frontend/src/store/bucketStore.ts

```diff
@@ -14,8 +21,19 @@ export const useBucketStore = defineStore('bucket', () => {
   async function load() {
     try {
       loading.value = true;
-      const response = await bucketService.searchForBuckets();
-      buckets.value = response.data;
+
+      const identityId = getKeycloak.value.tokenParsed?.idir_user_guid;
+      if (identityId) {
+        await userStore.searchUsers({ identityId: identityId });
+
+        if (userSearch.value.length) {
+          const userId = userSearch.value[0].userId;
+          const permResponse = (await bucketService.searchForPermissions({ userId, objectPerms: true })).data;
+          const uniqueIds = [...new Set(permResponse.map((x: any) => x.bucketId))];
```

Comment:
Can this be typecasted into x: Object or will it complain if we did that?

Reply:
It will complain that bucketId does not exist on type Object.

Reply:
Try embedding a direct in-line type then:
```ts
const uniqueIds = [...new Set(permResponse.map((x: { bucketId: string }) => x.bucketId))];
```